src/string.cr

```diff
@@ -1056,59 +1056,70 @@ class String
     end
   end
 
-  def unsafe_byte_at(index)
+  # Returns the byte at the given *index* without bounds checking.
+  def unsafe_byte_at(index : Int) : UInt8
     to_unsafe[index]
   end
 
-  # Returns a new `String` with each uppercase letter replaced with its lowercase
-  # counterpart.
+  # Returns a new `String` with each uppercase letter replaced with its lowercase counterpart.
   #
   # ```
   # "hEllO".downcase # => "hello"
   # ```
-  def downcase(options = Unicode::CaseOptions::None)
+  def downcase(options : Unicode::CaseOptions = :none) : String
     return self if empty?
 
+    String.build(bytesize) { |io| downcase io, options }
```

Comment:
This is losing the optimization that for ascii we know exactly the size. `String.build` is a bit slower than `String.new`.